pkg/controller/datadogagentdeployment/secret.go

```diff
@@ -149,3 +150,40 @@ func needClusterAgentSecret(dad *datadoghqv1alpha1.DatadogAgentDeployment) bool
 
 	return false
 }
+
+// getCredentials returns the Datadog API Key and APP Key, it returns an error if one key is missing
+func (r *ReconcileDatadogAgentDeployment) getCredentials(dad *datadoghqv1alpha1.DatadogAgentDeployment) (string, string, error) {
+	apiKey, appKey := "", ""
+	if dad.Spec.Credentials.APIKey != "" {
+		apiKey = dad.Spec.Credentials.APIKey
+	} else {
+		secretName := getAPIKeySecretName(dad)
+		secret := &corev1.Secret{}
+		err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: dad.Namespace, Name: secretName}, secret)
+		if err != nil {
+			return "", "", err
+		}
+		apiKey = string(secret.Data[datadoghqv1alpha1.DefaultAPIKeyKey])
+	}
+	if dad.Spec.Credentials.AppKey != "" {
+		appKey = dad.Spec.Credentials.AppKey
+	} else {
+		secretName := getAppKeySecretName(dad)
+		secret := &corev1.Secret{}
+		err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: dad.Namespace, Name: secretName}, secret)
+		if err != nil {
+			return "", "", err
+		}
+		appKey = string(secret.Data[datadoghqv1alpha1.DefaultAPPKeyKey])
+	}
```

Comment:
maybe you can refactor the code with a sub function that take in parameters: `dad` and the function `getAppKeySecretName ` or `getAPIKeySecretName `